flow/project.py

```diff
@@ -868,11 +1043,151 @@ def add_operation(self, name, operation, directives=None):
+                                     "".format(name, self.name))
+
+        # Adding `select` parameters for a group.
+        # This parameter must be the same for every operation in a group
+        if len(self.operations) == 1:
+            self.flow_select = flow_select
+        else:
+            closure_consts = []
+            closure_self_consts = []
+            closure_freevars = []
+            closure_self_freevars = []
+
+            if flow_select.args[0] is not None:
+                if flow_select.args[0].__closure__ is not None:
+                    for cl in flow_select.args[0].__closure__:
+                        try:
+                            closure_consts.append(cl.cell_contents.__code__.co_consts)
+                        except Exception:
+                            closure_consts.append(cl.cell_contents)
+                        try:
+                            closure_freevars.append(cl.cell_contents.__code__.co_freevars)
+                        except Exception:
+                            closure_freevars.append(cl.cell_contents)
+
+                if self.flow_select.args[0].__closure__ is not None:
+                    for self_cl in self.flow_select.args[0].__closure__:
+                        try:
+                            closure_self_consts.append(self_cl.cell_contents.__code__.co_consts)
+                        except Exception:
+                            closure_self_consts.append(self_cl.cell_contents)
+                        try:
+                            closure_self_freevars.append(self_cl.cell_contents.__code__.co_freevars)
+                        except Exception:
+                            closure_self_freevars.append(self_cl.cell_contents)
+
+            if (
+                closure_consts != closure_self_consts or
+                closure_freevars != closure_self_freevars
+            ):
+                raise ValueError("Can't add the operation '{}' to the group '{}' "
+                                 "due to different select parameters.".
+                                 format(name, self.name))
+
+            try:
+                if (
+                    self.flow_select.args[0].__code__.co_consts !=
+                    flow_select.args[0].__code__.co_consts or
+                    self.flow_select.args[0].__code__.co_freevars !=
+                    flow_select.args[0].__code__.co_freevars
+                ):
+                    raise ValueError("Can't add the operation '{}' to the group '{}' "
+                                     "due to different select parameters.".
+                                     format(name, self.name))
+            except Exception:
+                if not (self.flow_select.args[0] == flow_select.args[0]):
+                    raise ValueError("Can't add the operation '{}' to the group '{}' "
+                                     "due to different select parameters.".
+                                     format(name, self.name))
```

Comment:
Why are we in a try block to raise an exception we immediately catch and reraise the same error? Does the conditional occasionally raise errors?